### PR TITLE
add transaction relayer to Kakarot EOA for tests

### PR DESCRIPTION
**Pull request by tcoratger**
Related https://github.com/kkrt-labs/kakarot-rpc/issues/1310

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,6 +58,9 @@ test: katana-genesis load-env
 test-target: load-env
 	cargo test --tests --all-features $(TARGET) -- --nocapture
 
+test-target1: load-env
+	RUST_LOG=info cargo test --package kakarot-rpc --test entry --all-features -- tests::eth_provider::test_send_raw_transaction --exact --show-output
+
```
Comment (greged93) [inline Makefile:63]: remove?

File: src/providers/eth_provider/starknet/relayer.rs [hunk 2]
```diff
@@ -28,24 +27,24 @@ static RELAYER_SIGNER: LazyLock<LocalWallet> = LazyLock::new(|| {
 /// A relayer holding a lock on a mutex on an account and connected to the Starknet network.
 /// The relayer is used to sign  transactions and broadcast them on the network.
 #[derive(Debug)]
-pub struct LockedRelayer<'a> {
+pub struct LockedRelayer<'a, SP: Provider + Send + Sync> {
```
Comment (greged93) [inline src/providers/eth_provider/starknet/relayer.rs:30]: do we need SP here? I don't think we will ever use something else than `JsonRpcClient<HttpTransport>`

File: tests/tests/eth_provider.rs [hunk 3]
```diff
@@ -747,6 +755,35 @@ async fn test_send_raw_transaction(#[future] katana: Katana, _setup: ()) {
         .await
         .expect("failed to send transaction");
 
+    // Prepare the relayer
+    let relayer = katana.sequencer.account();
```
Comment (greged93) [inline tests/tests/eth_provider.rs:759]: should we maybe have the relayer be a field of the EOA? then you can get a relayer easily and just need to create the relayer at creation of the EOA instance?

File: src/providers/eth_provider/starknet/relayer.rs (hunk 2 above)
Comment (tcoratger) [reply src/providers/eth_provider/starknet/relayer.rs:30]: Mhhh I think so, maybe once this is stable we can have a look at how to simplify but in the `deploy_evm_contract`, I had to use:

```rust
// Relay the transaction
        let starknet_transaction_hash = LockedRelayer::new(
            current_nonce.lock().await,
            relayer.address(),
            relayer_balance,
            self.starknet_provider(),
            self.starknet_provider().chain_id().await.expect("Failed to get chain id"),
        )
        .relay_transaction(&tx_signed)
        .await
        .expect("Failed to relay transaction");
```

which involved the starknet provider which is SP, that's why I had to put a generic here...

File: tests/tests/eth_provider.rs (hunk 3 above)
Comment (tcoratger) [reply tests/tests/eth_provider.rs:759]: Yeah good idea just fixed

File: Makefile (hunk 1 above)
Comment (tcoratger) [reply Makefile:63]: oups yes just removed

**Review by greged93 — APPROVED** [review]

**Review by ClementWalter — APPROVED** [review]

**Comment by greged93** [comment]
I think you need to bump kakarot submodule

**Comment by tcoratger** [comment]
@greged93 Yes indeed, in one of my recent commits the submodule must have been downgraded for some reason, probably a mismatch... Anyway, just bumped again

**Review by greged93 — APPROVED** [review]

**Pull request merged**